tests/core/pyspec/eth2spec/test/altair/unittests/test_sync_protocol.py

```diff
@@ -70,11 +73,16 @@ def test_process_light_client_update_not_timeout(spec, state):
     assert store.finalized_header == pre_store.finalized_header
     assert store.best_valid_update == update
 
+    spec.config = old_config
+
 
 @with_altair_and_later
 @spec_state_test
 @with_presets([MINIMAL], reason="too slow")
 def test_process_light_client_update_at_period_boundary(spec, state):
+    old_config = spec.config
+    override_config_fork_epochs(spec, state)
```

Comment:
Since we want to transform these tests to test vectors, the override config has to be yielded to test vectors.
I'd suggest using `spec_configured_state_test` decorator that would yield the overridden configs:

1. Write a new *decorator* `override_config_fork_epoch` that call `spec_configured_state_test`
    - Usage example: https://github.com/ethereum/consensus-specs/pull/2767/files 
2. Put this decorator to all these sync protocol test cases

Reply:
Managed to create a decorator that applies the config correctly :) And added a test for the decorator as well.